### PR TITLE
Update dependency go modules for k8s v1.27.0-rc.0

### PR DESCRIPTION
**Pull request by sunnylovestiramisu**
Ran kubernetes-csi/csi-release-tools go-get-kubernetes.sh -p 1.27.0-rc.0

```release-note
NONE
```

### CODE REVIEW

**Comment by sunnylovestiramisu** [comment]
The https://github.com/kubernetes-csi/livenessprobe/blob/62c447bb251bcedb31a1e2203903130d5414c4e9/.github/workflows/trivy.yaml is running on a deprecating ubuntu version, we need to update this as well.

**Comment by sunnylovestiramisu** [comment]
@andyzhangx Do you have any idea which version of ubuntu we should use next?

**Comment by andyzhangx** [comment]
> @andyzhangx Do you have any idea which version of ubuntu we should use next?

@sunnylovestiramisu this PR would fix the issue: https://github.com/kubernetes-csi/livenessprobe/pull/179, and the trivy github action detects one CVE, I also fixed that CVE in the PR.

File: go.mod [hunk 1]
```diff
@@ -28,6 +28,6 @@ require (
 	golang.org/x/text v0.8.0 // indirect
 	google.golang.org/genproto v0.0.0-20220804142021-4e6b2dfa6612 // indirect
 	google.golang.org/protobuf v1.28.1 // indirect
-	k8s.io/apimachinery v0.26.0 // indirect
+	k8s.io/apimachinery v0.27.0-rc.0 // indirect
 	k8s.io/component-base v0.26.0 // indirect
```
Comment (xing-yang) [inline go.mod:32]: Why is this one v0.26.0, not 0.27.0-rc.0?

Comment (pohly) [reply go.mod:32]: Why update to a release candidate at all?

1.27 is going to be out next week.

Comment (sunnylovestiramisu) [reply go.mod:32]: The script automatically picked this version.

**Pull request merged**